lib/web/ui/integration.go

```diff
@@ -181,6 +212,13 @@ func (r *UpdateIntegrationRequest) CheckAndSetDefaults() error {
 			return trace.Wrap(err)
 		}
 	}
+	if r.OAuth != nil {
+		// Update allows reuse of the existing ID but secret must always be
+		// provided.
+		if r.OAuth.Secret == "" {
+			return trace.BadParameter("missing OAuth secret for GitHub integration")
+		}
+	}
```

Comment:
should we also add check for empty `r.OAuth.ID` too? since the api allows it to be modified anyways?